testrepository/tests/test__compute_context.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (c) 2015 Testrepository Contributors
+#
+# Licensed under either the Apache License, Version 2.0 or the BSD 3-clause
+# license at the users choice. A copy of both licenses are available in the
+# project source as Apache-2.0 and BSD. You may not use this file except in
+# compliance with one of these two licences.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under these licenses is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# license you chose for the specific language governing permissions and
+# limitations under that license.
+
+"""Tests for the _computecontext module."""
+
+from __future__ import unicode_literals
+
+from testrepository._computecontext import (
+    Cache,
+    Instance,
+    )
+from testrepository.tests import ResourcedTestCase
+
+class TestInstance(ResourcedTestCase):
+
+    def test_fields(self):
+        instance = Instance('profile', '1')
+        self.assertEqual('profile', instance.profile)
+        self.assertEqual('1', instance.id)
+
+    def test_can_be_put_in_set(self):
+        instances = set()
+        instances.add(Instance('profile', '2'))
+
+    def test_rejects_bytes(self):
+        self.assertRaises(ValueError, Instance, 'profile', b'1')
+        self.assertRaises(ValueError, Instance, b'profile', '1')
+
+
+class TestCache(ResourcedTestCase):
```

Comment:
The `Cache` would be an almost perfect place to use Hypothesis, and specifically its [stateful testing](http://hypothesis.readthedocs.org/en/latest/stateful.html) APIs.